### ISSUE TITLE
Modal did not disappear when its parent unmount.

### ISSUE THREAD

**Issue by wathui99** [jacklam718/react-native-modals#227]
As the title, Modal did not disappear when its parent unmount. This happens when I click save -> a new modal will render and I can't close the old modal. It still there!
I suggest add componentWillUnmount to solve this:
```
key: 'componentWillUnmount',
value: function componentWillUnmount() {
  if(this.sibling !== null) {
    this.sibling.destroy();
  }
}
```

**Comment by wesleylopex** [comment]
I am with the same problem, any resolution?

**Comment by wathui99** [comment]
As I suggested, add lines code above into file <module folder>/dist/Modal.js

**Comment by wesleylopex** [comment]
is it happening when you click on the hardware back button?
the Modal has a prop called onHardwareBackPress you pass the function to close the modal, this is the solve i found, might help you

**Comment by wesleylopex** [comment]
This is my modal configuration

<BottomModal
      visible={isModalVisible}
      modalStyle={styles.modal}
      modalAnimation={
        new SlideAnimation({
          initialValue: 0,
          slideFrom: "bottom",
          useNativeDriver: true,
        })
      }
      onSwipeOut={() => props.hideModal()}
      onTouchOutside={() => props.hideModal()}
      onHardwareBackPress={() => props.hideModal()}
    >

**Comment by choudharyjji** [comment]
using onHardwareBackPress throw me an error " TypeError: undefined is not a function"

**Comment by mbrehme** [comment]
@wathui99 your solution solve some issues for me too. Would be cool to have this added!

**Comment by jacklam718** [comment]
@wathui99 Thanks for the suggestion. This issue has been fixed in `v0.22.1`
